src/SideNavigation/SideNavigation.js

```diff
@@ -1,204 +1,263 @@
+                onClick={(e) => {
+                    onClick(e);
+                    onItemSelect(e, id, hasChild);
+                }}>
+                {glyph ? (
+                    <span
+                        className={`fd-side-nav__icon sap-icon--${glyph} sap-icon--l`}
+                        role='presentation' />
+                ) : null}
+                {name}
+            </a>
+        );
+    };
+
+    let hasChild = false;
+    React.Children.forEach(children, (child) => {
+        if (React.isValidElement(child) && child.type === SideNavSubList) {
+            hasChild = true;
+        }
+    });
+
+    return (
+        <li {...props}
+            className='fd-side-nav__item'
+            key={id}>
+            {url && renderLink()}
+            {React.Children.map(children, (child) => {
+                if (React.isValidElement(child) && child.type !== SideNavSubList) {
+                    return React.cloneElement(child, {
+                        children: (<React.Fragment>
+                            {glyph ? (
+                                <span
+                                    className={`fd-side-nav__icon sap-icon--${glyph} sap-icon--l`}
+                                    role='presentation' />
+                            ) : null}
+                            {child.props.children}
+                        </React.Fragment>),
+                        className: getClasses()
+                    });
+                } else if (React.isValidElement(child) && child.type === SideNavSubList) {
+                    return React.cloneElement(child, {
+                        onItemSelect: onItemSelect,
+                        open: expandedIds.includes(id),
+                        selectedId: selectedId
+                    });
+                } else {
+                    return child;
+                }
+            })}
+        </li>
     );
+};
 
+SideNavListItem.propTypes = {
+    children: PropTypes.node,
+    expandedIds: PropTypes.array,
+    glyph: PropTypes.string,
+    id: PropTypes.string,
+    isSubItem: PropTypes.bool,
+    name: PropTypes.string,
+    selected: PropTypes.bool,
+    selectedId: PropTypes.string,
+    url: PropTypes.string,
+    onClick: PropTypes.func,
+    onItemSelect: PropTypes.func
+};
+
+SideNavListItem.propDescriptions = {
+    expandedIds: 'An array of strings for expanded `SideNavListItems`\'s.',
+    isSubItem: 'Set to **true** to display as a subitem.',
+    name: 'Link text to be set in conjunction with the `url` prop.',
+    onItemSelect: 'Callback function when user selects an item .',
+    selected: 'Set to **true** to display as selected.',
+    selectedId: 'The id of the selected `SideNavListItem`.',
+    url: 'Creates an internal anchor when a child anchor is not provided.'
+};
+
+SideNavListItem.defaultProps = {
+    onClick: () => {}
+};
+
+export const SideNavSubList = ({children, id, onItemSelect, open, selectedId, ...props}) => {
```

Comment:
Do we need this extra component?  Semantically, isn't a sub list just another list?  Maybe have an additional prop like `hasParent` (that could be passed in the `cloneElement` calls) that allows you to add different classes or do other conditional logic.

Reply:
This component has been refactored out.

Reply:
it's still being exported. should it?